akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala

```diff
@@ -72,12 +85,25 @@ private[akka] trait DeathWatch { this: ActorCell ⇒
     watching.contains(subject) || (subject.path.uid != ActorCell.undefinedUid &&
       watching.contains(new UndefinedUidActorRef(subject)))
 
+  // TODO this should be removed and be replaced with `watching.get(subject)`
+  //   when all actor references have uid, i.e. actorFor is removed
+  private def watchingGet(subject: ActorRef): Option[Option[Any]] =
```

Comment:
When can these helpers be removed? AFAICT `actorFor` has been deprecated since 2.2, so we could've removed it in 2.4.0 (perhaps 2.3.0?), but since we didn't, it will live on until 3.0.0?